src/aggregation/agg_req.rs

```diff
@@ -102,10 +102,25 @@ pub(crate) struct BucketAggregationInternal {
 impl BucketAggregationInternal {
     pub(crate) fn as_histogram(&self) -> &HistogramAggregation {
         match &self.bucket_agg {
-            BucketAggregationType::Range(_) => panic!("unexpected aggregation"),
             BucketAggregationType::Histogram(histogram) => histogram,
+            _ => panic!("unexpected aggregation"),
         }
     }
+    pub(crate) fn as_term(&self) -> &TermsAggregation {
```

Comment:
can we return an Option and panic on the call site?

Reply:
yep